src/lib/qnio/iioapi.c

```diff
@@ -481,7 +481,7 @@ iio_open(const char *uri, const char *devid, uint32_t flags)
     if (hostinfo == NULL) {
         nioDbg("Unable to read the host information for device %s\n", devid);
         hostinfo = (struct iio_vdisk_hostinfo *)malloc(sizeof (struct iio_vdisk_hostinfo));
-        strncpy(hostinfo->hosts[hostinfo->nhosts], uri, NAME_SZ);
+        strncpy(hostinfo->hosts[0], uri, NAME_SZ);
```

Comment:
A pre-existing problem:

This strncpy() call is unsafe.  If uri is at least NAME_SZ characters long then ->hosts[0] will lack the NUL-terminator.  Later calls to string functions may access memory beyond the end of ->hosts[0].

This is a well-known issue with strncpy() and many coding standards forbid using this function.  Here is one reference that covers the problem:
https://www.securecoding.cert.org/confluence/display/c/STR32-C.+Do+not+pass+a+non-null-terminated+character+sequence+to+a+library+function+that+expects+a+string

It's probably best to eliminate strncpy() throughout the code to avoid these bugs.  snprintf(hostinfo->hosts[0], sizeof(hostinfo->hosts[0]), "%s", uri) solves the NUL-terminator issue.  You still need to handle truncation (e.g. returning an error so the user knows the URI was too long) so that it's not silently truncated.